### PR TITLE
Avoid broken pytest-html plugin

### LINKED ISSUE #568: Unit tests failing because pytest-html 2.1.0 is broken
See pytest-dev/pytest-html#282

### PR DESCRIPTION
**Pull request by bouweandela**
Before you start, please read [CONTRIBUTING.md](https://github.com/ESMValGroup/ESMValCore/blob/master/CONTRIBUTING.md).

**Tasks**

-   [x] [Create an issue](https://github.com/ESMValGroup/ESMValCore/issues) to discuss what you are going to do, if you haven't done so already (and add the link at the bottom)
-   [x] This pull request has a descriptive title that can be used in a changelog
-   [ ] Circle/CI tests pass. Status can be seen below your pull request. If the tests are failing, click the link to find out why.
-   [ ] Codacy code quality checks pass. Status can be seen below your pull request. If there is an error, click the link to find out why. If you suspect Codacy may be wrong, please ask by commenting.

* * *

Closes #568

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -51,7 +51,7 @@
         'pytest-cov',
         'pytest-env',
         'pytest-flake8',
-        'pytest-html',
+        'pytest-html!=2.1.0',
```
Comment (valeriupredoi) [inline setup.py]:
```suggestion
        'pytest-html<2.1.0',
```

Comment (valeriupredoi) [reply setup.py]: a bit more flexible for the environment

Comment (bouweandela) [reply setup.py]: I would prefer specifically avoiding a single version, from looking at the discussion in the issue I suspect the issue will be solved with the next release.

**Comment by valeriupredoi** [comment]
@bouweandela you totally ignored my suggestion, how did you manage to merge?

**Comment by valeriupredoi** [comment]
also, any reason why in `meta.yaml`:
```
  run:
    # esmvaltool
    - python>=3.6,<3.8
```

**Comment by mattiarighi** [comment]
> @bouweandela you totally ignored my suggestion, how did you manage to merge? Correction: @mattiarighi

Sorry, I didn't think it was critical and @bouweandela reply looked reasonable.

File: setup.py (hunk 1 above)
Comment (valeriupredoi) [reply setup.py]: yes, but conda does prefer `<` to `!=`, anyways this is pypi so should be cool

**Comment by valeriupredoi** [comment]
yea all cool, am a bit pedantic when it comes to the environment. @bouweandela what's up with the pin on python still in `meta.yaml`, dude?

**Comment by bouweandela** [comment]
Forgot about it, but I though you fixed that? Or was that for the other repo?

**Pull request merged**